spec/jobs/set_catalog_record_ids_and_barcodes_csv_job_spec.rb

```diff
@@ -45,172 +47,59 @@
+        [druids[0], 'superbad', catalog_record_ids[0], "#{catalog_record_id_prefix}55555", refresh[0]].join(','),
+        [druids[1], barcodes[1], 'trash', '', refresh[1]].join(','),
+        [druids[2], barcodes[2], catalog_record_ids[2], '', refresh[2]].join(',')
+      ].join("\n")
     end
 
-    it 'attempts to update the catalog_record_id/barcode for each druid with correct corresponding catalog_record_id/barcode' do
+    it 'only attempts to update the catalog_record_id/barcode for the one druid with valid barcode/catalog_record_id' do
+      subject.perform(bulk_action.id, { csv_file: StringIO.new(csv) })
       expect(bulk_action.druid_count_total).to eq druids.length
-      expect(bulk_action.druid_count_fail).to eq 0
-      expect(subject).to have_received(:update_catalog_record_id_and_barcode)
-        .with(ItemChangeSet, Hash, buffer).exactly(druids.length).times
+      expect(bulk_action.druid_count_fail).to eq 2
     end
+  end
 
-    context 'with invalid barcode and catalog_record_ids' do
-      let(:csv_file) do
-        [
-          "Druid,Barcode,#{catalog_record_id_column},#{catalog_record_id_column},Refresh",
-          [druids[0], 'superbad', catalog_record_ids[0], "#{catalog_record_id_prefix}55555", refresh[0]].join(','),
-          [druids[1], barcodes[1], 'trash', '', refresh[1]].join(','),
-          [druids[2], barcodes[2], catalog_record_ids[2], '', refresh[2]].join(',')
-        ].join("\n")
-      end
-
-      it 'only attempts to update the catalog_record_id/barcode for the one druid with valid barcode/catalog_record_id' do
-        expect(bulk_action.druid_count_total).to eq druids.length
-        expect(bulk_action.druid_count_fail).to eq 2
-        expect(subject).to have_received(:update_catalog_record_id_and_barcode)
-          .with(ItemChangeSet, Hash, buffer).exactly(druids.length - 2).times
-      end
+  context 'when not authorized' do
+    let(:authorized_to_update) { false }
+
+    it 'logs and returns' do
+      subject.perform(bulk_action.id, { csv_file: StringIO.new(csv) })
+      expect(bulk_action.druid_count_total).to eq druids.length
+      expect(bulk_action.druid_count_fail).to eq 3
+      expect(buffer.string).to include('Not authorized')
     end
   end
 
-  describe '#update_catalog_record_id_and_barcode' do
-    let(:druid) { druids[0] }
-    let(:catalog_record_id_prefix) { 'a' }
-    let(:catalog_record_ids) { ["#{catalog_record_id_prefix}12345,#{catalog_record_id_prefix}66233", '', "#{catalog_record_id_prefix}44444"] } # ''
-    let(:catalog_record_ids_arg) { catalog_record_ids[0].split(',') }
-    let(:object_client) { instance_double(Dor::Services::Client::Object, update: true) }
-
+  context 'when error' do
     before do
-      allow(Dor::Services::Client).to receive(:object).with(druid).and_return(object_client)
-      allow(subject.ability).to receive(:can?).and_return(true)
+      allow(VersionService).to receive(:open?).and_raise('Oops')
     end
 
-    context 'when a DRO' do
```

Comment:
We were testing private methods before and a whole lot of implementation details.